File: cheongha/BinarySearch/고정점찾기.py
# p.368
# 2021-09-01 21:21-
# 5
# -15 -6 1 3 7

n = int(input())
array = list(map(int, input().split()))

def binary_search(array, target, start, end):
    while start <= end:
        mid = (start + end) // 2
        if array[mid] == target:
            return mid
        elif array[mid] > target:
            end = mid - 1
        else:
            start = mid + 1
    return None

for i in array:
    if binary_search(array, i, 0, n-1) == i:
        result = i
        break
    else:
        result = -1
print(result)

File: yezi/이진탐색/고정점 찾기.swift
//
//  고정점 찾기.swift
//  CodingTest
//
//  Created by 윤예지 on 2021/09/07.
//

import Foundation

let n = Int(readLine()!)!
var array = Array(readLine()!).split(separator: " ").map { Int(String($0))! }

func binary_search(array: [Int], start: Int, end: Int) -> Int {
    if start > end {
        return -1
    }
    var mid = (start + end) / 2
    if array[mid] == mid {
        return mid
    } else if array[mid] > mid {
        return binary_search(array: array, start: start, end: mid-1)
    } else {
        return binary_search(array: array, start: mid+1, end: end)
    }
}

print(binary_search(array: array, start: 0, end: n-1))
